### PR TITLE
feat(docs): add Dataverse card (was missing)

### PR DESCRIPTION
**Pull request by ccamel**
Self explanatory.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new card for the "Dataverse" Smart Contract in the API Reference documentation.
	- Added a brief description and a link to the Dataverse Smart Contract documentation.
	- Updated the architecture overview to include the new Dataverse card under "The Smart Contracts" section.

### CODE REVIEW

**Review by AnjaMis — APPROVED** [review]

**Review by amimart — APPROVED** [review]
Good catch! 👍

File: docs/api-reference/api-reference.mdx [hunk 1]
```diff
@@ -34,6 +34,14 @@ Deepen your understanding of the Axone Protocol's Smart Contracts.
         <p class="card-text">Introducing the Governance Interpreter.</p>
       </div>
     </div>
+    <div class="col">
+      <div class="card card-body h-100 d-flex flex-column">
+        <a href="/contracts/axone-dataverse" class="card-title card-link stretched-link">
```
[the remark this answers is not in the document]
Comment (ccamel) [reply docs/api-reference/api-reference.mdx:39]: Nope. Link is correct.

File: docs/architecture/overview.mdx [hunk 2]
```diff
@@ -67,6 +67,14 @@ Deepen your understanding of the Axone Protocol's Smart Contracts.
         <p class="card-text">Introducing the Governance Interpreter.</p>
       </div>
     </div>
+    <div class="col">
+      <div class="card card-body h-100 d-flex flex-column">
+        <a href="/contracts/axone-dataverse" class="card-title card-link stretched-link">
```
[the remark this answers is not in the document]
Comment (ccamel) [reply docs/architecture/overview.mdx:72]: Nope. Link is correct.

**Comment by ccamel** [comment]
@coderabbitai resolve

**Pull request merged**